crates/integration_test/src/tests/cgroups/blkio.rs

```diff
@@ -0,0 +1,578 @@
+    }
+
+    // we assume that is read_iops is supported, write_iops is also supported
+    if supports_throttle_iops() {
+        // throttle read iops
+
+        let throttle_read_iops_string =
+            fs::read_to_string(format!("{}/blkio.throttle.read_iops_device", path))
+                .context("error in reading block io read iops device")?;
+        let mut throttle_devices = Vec::new();
+        for line in throttle_read_iops_string.lines() {
+            let (device_data, rate) = line
+                .split_once(" ")
+                .context("invalid throttle device format")?;
+            let (major_str, minor_str) = device_data
+                .split_once(":")
+                .context("invalid major-minor number format")?;
+            throttle_devices.push(ThrottleDevice {
+                major: major_str.parse().context("error in parsing major number")?,
+                minor: minor_str.parse().context("error in parsing minor number")?,
+                rate: rate
+                    .parse()
+                    .context("error in parsing throttle read iops rate")?,
+            });
+        }
+        device.throttle_read_iops_devices = throttle_devices;
+
+        // throttle write iops
+
+        let throttle_write_iops_string =
+            fs::read_to_string(format!("{}/blkio.throttle.write_iops_device", path))
+                .context("error in reading block io write iops device")?;
+        let mut throttle_devices = Vec::new();
+        for line in throttle_write_iops_string.lines() {
+            let (device_data, rate) = line
+                .split_once(" ")
+                .context("invalid throttle device format")?;
+            let (major_str, minor_str) = device_data
+                .split_once(":")
+                .context("invalid major-minor number format")?;
+            throttle_devices.push(ThrottleDevice {
+                major: major_str.parse().context("error in parsing major number")?,
+                minor: minor_str.parse().context("error in parsing minor number")?,
+                rate: rate
+                    .parse()
+                    .context("error in parsing throttle write iops rate")?,
+            });
+        }
+        device.throttle_write_iops_devices = throttle_devices;
+    }
+
+    Ok(device)
+}
+
+/// validates the BlockIO structure parsed from /sys/fs/cgroup/blkio
+/// with the spec
+fn validate_block_io(cgroup_name: &str, spec: &Spec) -> Result<()> {
+    let cgroup_path = PathBuf::from(CGROUP_ROOT)
+        .join("blkio/runtime-test")
+        .join(cgroup_name);
+    let block_io = get_blkio_data(cgroup_path.to_str().unwrap())?;
+
+    let resources = spec.linux().as_ref().unwrap().resources().as_ref().unwrap();
+    let spec_block_io = resources.block_io().as_ref().unwrap();
+    // weight ------
+    if supports_weight() {
+        if spec_block_io.weight().is_none() {
+            return Err(anyhow::anyhow!("spec block io weight is none"));
```

Comment:
```suggestion
            bail!("spec block io weight is none");
```
A few more instances of this.